src/Blueprint/Blueprint/Cmdlets/BlueprintCmdletBase.cs

```diff
@@ -0,0 +1,360 @@
+            }
+            set => graphRbacManagementClient = value;
+        }
+
+        /// <summary>
+        /// Authorization client
+        /// </summary>
+        private IAuthorizationManagementClient authorizationManagementClient;
+
+        public IAuthorizationManagementClient AuthorizationManagementClient
+        {
+            get
+            {
+                return authorizationManagementClient = authorizationManagementClient ?? AzureSession.Instance.ClientFactory.CreateArmClient<AuthorizationManagementClient>(DefaultProfile.DefaultContext, 
+                                                           AzureEnvironment.Endpoint.ResourceManager);
+            }
+            set => authorizationManagementClient = value;
+        }
+
+        /// <summary>
+        /// ARM client
+        /// </summary>
+        private IResourceManagementClient resourceManagerClient;
+        public IResourceManagementClient ResourceManagerClient
+        { 
+            get
+            {
+                return resourceManagerClient = resourceManagerClient ?? new ResourceManagementClient(
+                                                   DefaultProfile.DefaultContext.Environment.GetEndpointAsUri(AzureEnvironment.Endpoint.ResourceManager),
+                                                   ClientCredentials);
+            }
+            set => resourceManagerClient = value;
+        }
+
+        #endregion Properties
+
+        #region Cmdlet Overrides
+        protected override void WriteExceptionError(Exception ex)
+        {
+            var aggEx = ex as AggregateException;
+
+            if (aggEx != null && aggEx.InnerExceptions != null)
+            {
+                foreach (var e in aggEx.Flatten().InnerExceptions)
+                {
+                    WriteExceptionError(e);
+                }
+                return;
+            }
+            throw ex;
+        }
+        #endregion Cmdlet Overrides
+
+        #region Protected Methods
+
+        protected void ThrowIfAssignmentExits(string scope, string name)
+        {
+            PSBlueprintAssignment assignment = null;
+
+            assignment = BlueprintClient.GetBlueprintAssignment(scope, name);
+
+            if (assignment != null)
+            {
+                throw new Exception(string.Format(Resources.AssignmentExists, name, scope));
+            }
+        }
+
+        protected void ThrowIfAssignmentNotExist(string scope, string name)
+        {
+            PSBlueprintAssignment assignment = null;
+
+            try
+            {
+                assignment = BlueprintClient.GetBlueprintAssignment(scope, name);
+            }
+            catch (Exception ex)
+            {
+                if (ex is CloudException cex && cex.Response.StatusCode != System.Net.HttpStatusCode.NotFound)
+                {
+                    // if exception is for a reason other than .NotFound, pass it to the caller.
+                    throw ex;
```

Comment:
>throw ex; [](start = 20, length = 9)

If you throw an exception that was already thrown somewhere else, it wipes out the original stack trace. (We might need it for debugging sometime.) You should use "throw;" instead.